### PR TITLE
Petty Cash Expenses and Transactions Management.

### PR DESCRIPTION
**Pull request by mhamzajutt96**
Transactions Management of all the Users and their expenses.

### CODE REVIEW

File: app/controllers/admin_panel_controller.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class AdminPanelController < ApplicationController
   load_and_authorize_resource class: false
   
   def index
+    @expenses = current_user.transactions.map { |t| t.amount }.sum + current_user.expenses.map{ |e| e.amount }.sum
```
Comment (saqib1324) [inline app/controllers/admin_panel_controller.rb]: this logic belongs to the User model. you can name the method `all_expenses`

File: app/controllers/transactions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class TransactionsController < ApplicationController
+
+  def create
+    @transaction = Transaction.new(transaction_params)
+    if params[:transaction][:receiver_id].present?
```
Comment (saqib1324) [inline app/controllers/transactions_controller.rb]: what about `transaction_params(:receiver_id)`

File: app/controllers/user_panel_controller.rb [hunk 3]
```diff
@@ -2,6 +2,16 @@ class UserPanelController < ApplicationController
 	
 	load_and_authorize_resource class: false
 
+	def index
+		@expenses = current_user.transactions.map { |t| t.amount }.sum + current_user.expenses.map{ |e| e.amount }.sum
```
Comment (saqib1324) [inline app/controllers/user_panel_controller.rb]: as said earlier this logic belongs to model

File: app/controllers/user_panel_controller.rb [hunk 4]
```diff
@@ -2,6 +2,16 @@ class UserPanelController < ApplicationController
...
+		@receiveables = Transaction.where(receiver_id: current_user.id).map { |t| t.amount }.sum
+		if (@expenses - @receiveables) > 0
+      @message = "You need to receive #{@expenses - @receiveables} amount from admin."
```
Comment (saqib1324) [inline app/controllers/user_panel_controller.rb]: include abs function

File: app/views/categories/_form.html.erb [hunk 5]
```diff
@@ -12,8 +12,8 @@
     <%= form.text_field :name, autofocus: true, placeholder: 'Enter Name of Category', class: 'form-control login-form', required: true %>
   </div>
   <%= form.fields_for :automation_keywords do |keyword| %>
-    <%= keyword.text_field :keyword1, :value => category.automation_keywords[:keyword1], placeholder: 'Enter first keyword', class: 'form-control login-form', required: true %>
-    <%= keyword.text_field :keyword2, :value => category.automation_keywords[:keyword2], placeholder: 'Enter second keyword', class: 'form-control login-form', required: true %>
+    <%= keyword.text_field :keyword1, :value => (category.automation_keywords[:keyword1] if category.automation_keywords?), placeholder: 'Enter first keyword', class: 'form-control login-form', required: true %>
+    <%= keyword.text_field :keyword2, :value => (category.automation_keywords[:keyword2] if category.automation_keywords?), placeholder: 'Enter second keyword', class: 'form-control login-form', required: true %>
```
Comment (saqib1324) [inline app/views/categories/_form.html.erb]: `category&.automation.....`

File: app/views/user_panel/index.html.erb [hunk 6]
```diff
@@ -10,7 +10,7 @@
             <div class="col-xl-10 col-lg-9 col-md-8 col-sm-6 col-7">
               <div class="row">
                 <div class="col-12">
-                  <h2 class="text-primary mb-3"> <%= current_user.name %> </h2>
+                  <h2 class="text-primary mb-3"> <%= ((current_user.name if current_user.name?) || current_user.email) %> </h2>
```
Comment (saqib1324) [inline app/views/user_panel/index.html.erb]: `current_user&.name || current_user&.email`

File: app/controllers/transactions_controller.rb (hunk 2 above)
Comment (mhamzajutt96) [reply app/controllers/transactions_controller.rb]: transaction_params[:receiver_id]

File: app/views/categories/_form.html.erb (hunk 5 above)
Comment (mhamzajutt96) [reply app/views/categories/_form.html.erb]: This Doesn't work.

File: app/controllers/admin_panel_controller.rb [hunk 7]
```diff
@@ -4,6 +4,8 @@ class AdminPanelController < ApplicationController
   load_and_authorize_resource class: false
   
   def index
+    @expenses = current_user.all_expenses
+    @receiveables = Transaction.where(receiver_id: current_user.id).map { |t| t.amount }.sum
```
Comment (saqib1324) [inline app/controllers/admin_panel_controller.rb]: `.pluck(:amount)`

File: app/controllers/user_panel_controller.rb [hunk 8]
```diff
@@ -2,6 +2,16 @@ class UserPanelController < ApplicationController
 	
 	load_and_authorize_resource class: false
 
+	def index
+		@expenses = current_user.all_expenses
+		@receiveables = Transaction.where(receiver_id: current_user.id).map { |t| t.amount }.sum
+		if (@expenses - @receiveables) > 0
+      @message = "You need to receive #{(@expenses - @receiveables).abs} amount from admin."
```
Comment (saqib1324) [inline app/controllers/user_panel_controller.rb]: fix indentation

File: app/views/layouts/_receive_cash_modal.html.erb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+            <%= form.label :amount %>
+            <%= form.text_field :amount %>
+          </div>
+
+          <div class="field">
+            <%= form.label :description %>
+            <%= form.text_area :description %>
+          </div>
+          Select User you want to receive cash from:
+          <%= form.select :initiator_id, options_for_select(User.where.not(id: current_user.id).map { |u| [u.email, u.id]}) %>
```
Comment (saqib1324) [inline app/views/layouts/_receive_cash_modal.html.erb]: this query for users belongs to controller

File: app/views/layouts/_pay_cash_modal.html.erb [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            <%= form.label :amount %>
+            <%= form.text_field :amount %>
+          </div>
+
+          <div class="field">
+            <%= form.label :description %>
+            <%= form.text_area :description %>
+          </div>
+          Select User you want to pay:
+          <%= form.select :receiver_id, options_for_select(User.where.not(id: current_user.id).map { |u| [u.email, u.id]}) %>
```
Comment (saqib1324) [inline app/views/layouts/_pay_cash_modal.html.erb]: this query for users belongs to controller

File: app/controllers/user_panel_controller.rb [hunk 11]
```diff
@@ -1,16 +1,25 @@
...
     else
-      @message = "You have to pay #{(@expenses - @receiveables).abs} amount to admin."
+      @message = "You have to pay #{(@expenses - @receiveables).abs} amount to
+			admin."
     end
-	end
+  end
+
+	def activities
+    @activities = current_user.activities
```
Comment (saqib1324) [inline app/controllers/user_panel_controller.rb]: I guess it is not displaying indentation correctly here, but can you check indentation of this method

File: app/views/layouts/_pay_cash_modal.html.erb [hunk 12]
```diff
@@ -19,7 +19,7 @@
             <%= form.text_area :description %>
           </div>
           Select User you want to pay:
-          <%= form.select :receiver_id, options_for_select(User.where.not(id: current_user.id).map { |u| [u.email, u.id]}) %>
+          <%= form.select :receiver_id, options_for_select(current_user.other_users) %>
```
Comment (saqib1324) [inline app/views/layouts/_pay_cash_modal.html.erb]: I am seeing two modals and in both modals you are querying for `current_users.other_users`, you should pass these as a variable inside these partials, so if on any view these both partials are rendered we should query `other_users` only once and pass in to both modals.

File: config/routes.rb [hunk 13]
```diff
@@ -1,14 +1,19 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   root 'home#index'
   resources :transactions, only: [:index, :create]
   resources :categories
   resources :properties
   resources :expenses
-  get '/change_access/#{parameter}', to: 'admin_panel#change_access', as: :change_access
+  get '/change_access/#{parameter}', to: 'admin_panel#change_access',
```
Comment (saqib1324) [inline config/routes.rb]: This looks fine! but can you please add ## comments and separate routes into different blocks as you did in some other project.

File: app/controllers/user_panel_controller.rb (hunk 11 above)
Comment (mhamzajutt96) [reply app/controllers/user_panel_controller.rb]: Yes, it's working fine.

**Pull request merged**